data/core/src/commonMain/kotlin/com/tunjid/heron/data/repository/PostRepository.kt

```diff
@@ -389,21 +389,29 @@ internal class OfflinePostRepository @Inject constructor(
                 .getOrNull()
                 ?.let {
                     if (it.validationStatus !is CreateRecordValidationStatus.Valid) return@let
-                    upsertInteraction(
-                        partial = when (interaction) {
-                            is Post.Interaction.Create.Like -> PostViewerStatisticsEntity.Partial.Like(
-                                likeUri = it.uri.atUri.let(::GenericUri),
-                                postUri = interaction.postUri,
-                                viewingProfileId = authorId,
-                            )
+                    transactionWriter.inTransaction {
+                        upsertInteraction(
```

Comment:
Perfect. One more change. The existing `upsertInteraction` also uses `transactionWriter.inTransaction { }`. We need both writes to happen in the same transaction, so remove the use of  `transactionWriter.inTransaction { }` in the `upsertInteraction` method.

The only transaction should be the one here.